datastore/bolt/api_key_test.go

```diff
@@ -0,0 +1,303 @@
+		Role:   role,
+		Hash:   encodedKey,
+		Salt:   salt,
+	}
+
+	require.NoError(t, apiKeyRepo.CreateAPIKey(context.Background(), newApiKey))
+	apiKey, err := apiKeyRepo.FindAPIKeyByMaskID(context.Background(), newApiKey.MaskID)
+
+	require.NoError(t, err)
+	require.Equal(t, newApiKey.UID, apiKey.UID)
+	require.Equal(t, newApiKey.MaskID, apiKey.MaskID)
+
+	//Find Non Existent Mask ID
+	apiKey, err = apiKeyRepo.FindAPIKeyByMaskID(context.Background(), uuid.NewString())
+	require.ErrorIs(t, err, datastore.ErrAPIKeyNotFound)
+}
+
+func Test_FindAPIKeyByHash(t *testing.T) {
+	db, closeFn := getDB(t)
+	defer closeFn()
+
+	apiKeyRepo := NewApiRoleRepo(db)
+
+	maskID, salt, encodedKey, err := generateAPIKey()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	role := auth.Role{Type: "super_user"}
+
+	newApiKey := &datastore.APIKey{
+		UID:    uuid.New().String(),
+		MaskID: maskID,
+		Name:   "Api Key",
+		Type:   "test_api_key",
+		Role:   role,
+		Hash:   encodedKey,
+		Salt:   salt,
+	}
+
+	require.NoError(t, apiKeyRepo.CreateAPIKey(context.Background(), newApiKey))
+	apiKey, err := apiKeyRepo.FindAPIKeyByHash(context.Background(), newApiKey.Hash)
+
+	require.NoError(t, err)
+	require.Equal(t, newApiKey.UID, apiKey.UID)
+	require.Equal(t, newApiKey.Hash, apiKey.Hash)
+
+	//Find Non Existent Hash
+	apiKey, err = apiKeyRepo.FindAPIKeyByHash(context.Background(), uuid.NewString())
+	require.ErrorIs(t, err, datastore.ErrAPIKeyNotFound)
+}
+
+func Test_RevokeAPIKeys(t *testing.T) {
+	db, closeFn := getDB(t)
+	defer closeFn()
+
+	apiKeyRepo := NewApiRoleRepo(db)
+
+	maskID, salt, encodedKey, err := generateAPIKey()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	role := auth.Role{Type: "super_user"}
+
+	newApiKey := &datastore.APIKey{
+		UID:    uuid.New().String(),
+		MaskID: maskID,
+		Name:   "Api Key",
+		Type:   "test_api_key",
+		Role:   role,
+		Hash:   encodedKey,
+		Salt:   salt,
+	}
+
+	require.NoError(t, apiKeyRepo.CreateAPIKey(context.Background(), newApiKey))
+
+	ids := []string{newApiKey.UID}
+	require.NoError(t, apiKeyRepo.RevokeAPIKeys(context.Background(), ids))
+
+	//Find Deleted API Key
+	_, err = apiKeyRepo.FindAPIKeyByID(context.Background(), newApiKey.UID)
+	require.ErrorIs(t, err, datastore.ErrAPIKeyNotFound)
+}
+
+func Test_LoadAPIKeysPaged(t *testing.T) {
+	type ApiKey struct {
+		UID   string
+		Name  string
+		Count int
+	}
+
+	type Expected struct {
+		ApiKeyCount    int
+		paginationData datastore.PaginationData
+	}
+
+	tests := []struct {
+		name     string
+		pageData datastore.Pageable
+		apiKeys  []ApiKey
+		expected Expected
+	}{
+		{
+			name:     "Load API Keys Paged",
+			pageData: datastore.Pageable{Page: 1, PerPage: 3},
+			apiKeys: []ApiKey{
+				{
+					UID:   uuid.NewString(),
+					Name:  "Group 1",
+					Count: 10,
+				},
+			},
+			expected: Expected{
+				ApiKeyCount:    3,
+				paginationData: datastore.PaginationData{Total: 10, TotalPage: 4, Page: 1, PerPage: 3, Prev: 0, Next: 2},
+			},
+		},
+	}
```

Comment:
I don't think test tables are useful if you're only planning to add one test case